src/main/java/uk/org/tombolo/execution/DataExportEngine.java

```diff
@@ -101,4 +103,38 @@ private void importDatasource(ImporterMatcher forceImports, DatasourceRecipe dat
 				forceImports.doesMatch(datasourceSpec.getImporterClass())
 		);
 	}
+
+	private Importer initialiseImporter(String importerClass, String configFile) throws Exception {
+		Config importerConfiguration = null;
+		if (configFile != null && !"".equals(configFile)) {
+			importerConfiguration = ConfigUtils.loadConfig(
+					AbstractRunner.loadProperties("Configuration file", configFile));
+
+		}
+		return (Importer) Class.forName(importerClass).getDeclaredConstructor(Config.class).newInstance(importerConfiguration);
+	}
+
+	private String verifyProvider() throws Exception {
+		String validProvider = null;
+
+		File specificationFile = new File(DataExportRunner.executionSpecPath);
+		if (!specificationFile.exists()) return validProvider;
+		JSONReader reader = new JSONReader(specificationFile, new ArrayList<>(Arrays.asList("importerClass", "provider")));
+		reader.getData();
+		ArrayList<String> importers = reader.getTagValueFromAllSections("importerClass");
+		ArrayList<String> providers = reader.getTagValueFromAllSections("provider");
+
+		mainLoop:
+		for (String importer : importers) {
+			for (String provider : providers) {
+				Importer imp = initialiseImporter(importer, "");
+				if (!provider.equals(imp.getProvider().getLabel())) {
```

Comment:
Sorry I missed this failure in the first review:
This loop fails for me because it continues if it finds the right provider and fails the following iteration with a different provider for the same importer. I tested it with `src/main/resources/executions/examples/london-cycle-traffic-air-quality-lsoa.json`. Does it fail for you too?

Reply:
good catch, yeah it does, because in fields it doesn't know which importer class to look for and falls over now this kind of makes it hard to fix

Reply:
good catch, it does fail for me, i got what you mean. :)